src/shared/mod.rs

```diff
@@ -237,3 +242,56 @@ pub fn handle_save(
 
     Err("invalid textDocument/didSave request".to_string())
 }
+
+pub fn get_bucket(src: &str) -> String {
+    let file = Parser::new(src).parse_file("".to_string());
+    let mut result: Option<String> = None;
+    walk(
+        &create_visitor(&mut |n| {
```

Comment:
this is breaking from our current pattern of defining structs and implementing the visitor trait for them.

Reply:
I think this one is much better. But i am open for discussion

Reply:
I think its important to be consistent, also this is storing state as a closure rather than explicitly with a struct.

Reply:
why are we storing something useless?

Reply:
`result` is the state that I was referring to